presto-tpch/src/test/java/com/facebook/presto/tpch/TestTpchMetadata.java

```diff
@@ -0,0 +1,117 @@
+import static io.airlift.tpch.TpchTable.ORDERS;
+import static io.airlift.tpch.TpchTable.PART;
+import static io.airlift.tpch.TpchTable.PART_SUPPLIER;
+import static io.airlift.tpch.TpchTable.REGION;
+import static io.airlift.tpch.TpchTable.SUPPLIER;
+import static java.util.Arrays.stream;
+import static org.testng.Assert.assertEquals;
+
+public class TestTpchMetadata
+{
+    private static final double TOLERANCE = 0.01;
+
+    private final TpchMetadata tpchMetadata = new TpchMetadata("tpch");
+    private final ConnectorSession session = null;
+
+    @Test
+    public void testTableStats()
+    {
+        TpchMetadata.SCHEMA_NAMES.forEach(schema -> {
+            double scaleFactor = TpchMetadata.schemaNameToScaleFactor(schema);
+
+            testTableStats(schema, REGION, 5);
+            testTableStats(schema, NATION, 25);
+            testTableStats(schema, SUPPLIER, 10_000 * scaleFactor);
+            testTableStats(schema, CUSTOMER, 150_000 * scaleFactor);
+            testTableStats(schema, PART, 200_000 * scaleFactor);
+            testTableStats(schema, PART_SUPPLIER, 800_000 * scaleFactor);
+            testTableStats(schema, ORDERS, 1_500_000 * scaleFactor);
+            testTableStats(schema, LINE_ITEM, 6_000_000 * scaleFactor);
+        });
+    }
+
+    @Test
+    public void testTableStatsWithConstraints()
+    {
+        TpchMetadata.SCHEMA_NAMES.forEach(schema -> {
+            double scaleFactor = TpchMetadata.schemaNameToScaleFactor(schema);
+
+            testTableStats(schema, ORDERS, Constraint.alwaysFalse(), 0);
+            testTableStats(schema, ORDERS, constraint(ORDER_STATUS, "NO SUCH STATUS"), 0);
+            testTableStats(schema, ORDERS, constraint(ORDER_STATUS, "F"), 730_400 * scaleFactor);
+            testTableStats(schema, ORDERS, constraint(ORDER_STATUS, "O"), 733_300 * scaleFactor);
+            testTableStats(schema, ORDERS, constraint(ORDER_STATUS, "P"), 38_543 * scaleFactor);
+            testTableStats(schema, ORDERS, constraint(ORDER_STATUS, "F", "NO SUCH STATUS"), 730_400 * scaleFactor);
+            testTableStats(schema, ORDERS, constraint(ORDER_STATUS, "F", "O", "P"), 1_500_000 * scaleFactor);
+        });
+    }
+
+    private void testTableStats(String schema, TpchTable<?> table, double expectedRowCount)
+    {
+        testTableStats(schema, table, Constraint.alwaysTrue(), expectedRowCount);
+    }
+
+    private void testTableStats(String schema, TpchTable<?> table, Constraint<ColumnHandle> constraint, double expectedRowCount)
+    {
+        TpchTableHandle tableHandle = tpchMetadata.getTableHandle(session, new SchemaTableName(schema, table.getTableName()));
+        TableStatistics tableStatistics = tpchMetadata.getTableStatistics(session, tableHandle, constraint);
+
+        double actualRowCountValue = tableStatistics.getRowCount().getValue();
+        assertEquals(tableStatistics.getTableStatistics(), ImmutableMap.of("row_count", new Estimate(actualRowCountValue)));
```

Comment:
there's TableStatistics.ROW_COUNT_STATISTIC_KEY, but it's private. It's a test, so I wouldn't bother (it'll fail if s/b changes the key)